test/hotspot/jtreg/runtime/records/recordAttributeTest.java

```diff
@@ -73,8 +71,8 @@ public static void main(String... args) throws Throwable {
         // badly formed Record attribute. No exception should be thrown.
         Class newClass = Class.forName("oldRecordAttribute");
 
-        // Test that loading a class whose super class is not java.lang.Record
-        // ignores a badly formed Record attribute. No exception should be thrown.
-        newClass = Class.forName("superNotJLRecord");
+        // Test that loading a class causes the Record attribute to get parsed
+        // even if its super class is not java.lang.Record
+        runTest("superNotJLRecord", "Truncated class file");
```

Comment:
This simple "inversion" of the test is not sufficient. The original test was that a bad record attribute is ignored when not a subclass of j.l.Record. The modified test should be that a bad record attribute is never ignored. But in addition there should be a new test that a good record attribute in a class that does not subclass j.l.Record, is also okay.